design-patterns/domain-driven-design/pattern2/README.md

```diff
@@ -0,0 +1,205 @@
+Auth Option: "Vendia User". // Let's use Vendia User
+```
+
+**Node 2**
+```
+Node Name: "MarketingReadOnlyNode". // you can change the name.
+Node Description: "Node dedicated to Marketing Activities". // You can change the description.
+Cloud Server Provider: "AWS"
+Node Region:"us-east-1". //you can change it
+Auth Option: "Vendia User". // Let's use Vendia User
+```
+
+  **Step 3:** Provide the Uni schema. Copy and paste the Uni schema from the schema.json file in the schema folder.
+
+Let’s review new things we added to the schema that is different from the Pattern 1. 
+
+Check out the new `CustomerProfile` entity.
+
+
+```json
+"CustomerProfile": {
+           "description": "Customer Profile",
+           "type": "array",
+           "items": {
+               "type": "object",
+               "properties":...
+```
+
+
+Notice the `x-vendia-acls` we added at the top of the schema. You will use it to enforce READ/WRITE visibility at the record level by allowing the consumer to add/update a record with 'aclInput` property. You will notice the structure of the 'aclInput` property in the next section. The `x-vendia-acls` `type` property should match the entity names - `CustomerAccount` and `CustomerProfile` - you used in the schema.
+
+
+```json
+"x-vendia-acls": {
+       "CustomerAccountAcl": {
+           "type": "CustomerAccount"
+       },
+       "CustomerProfileAcl": {
+           "type": "CustomerProfile"
+       }
+   }
+
+```
+
+You will specify the `aclInput` when you insert a new record.  We will use the below script in the GraphQL add command later.
+
+```graphql
+aclInput: {
+      acl: [
+        {
+          principal: {
+            nodes: [ "*" ]
+          },
+          operations: [ READ ]
+        }
+      ]
+    }
+
+```
+
+Notice the `customerId` property  and how it links `CustomerAccount` and `CustomerProfile`. You can use this pattern to avoid the eager loading of two separate aggregates. You don’t have to load both `CustomerProfile` and `CustomerAccount` at the same time.
+
+
+```json
+"CustomerProfile": {
+           "description": "Customer Profile",
+           "type": "array",
+           "items": {
+               "type": "object",
+               "properties": {
+                   "customerId": {
+                       "description": "The unique identifier for a customer and used as Forgien Key between CustomerAccout and CustomerProfile",
+                       "type": "string"
+                   }
+
+```
+
+Notice `zipcode` in `CustomerAccount{address{zipcode}` and `DMAbyZip` in the `CustomerProfile`. Designated Marketing Area(DMA) is marketing geo that is identified by the zipcode of the location. When a customer moves, the CRM Team updates the customer address in the ‘CustomerAccount{address{zipcode}} property. You can use `vendia_transaction` tag in the GraphQL command to update both the `zipcode` and `DMAbyZip` in the same transaction. You will try it later.
```

Comment:
Add a link to the Vendia Transactions docs

Reply:
added